File: count-the-number-of-consistent-strings1684/leetcode1684.py
class Solution:
    def countConsistentStrings(self, allowed, words):
        consistent = 0
        for word in words:
            for w in word :
                if w not in allowed:
                    break
            else:
                consistent+=1
        return consistent

allowed = "ab"
words = ["ad","bd","aaab","baa","badab"]
obj = Solution()
print(obj.countConsistentStrings(allowed, words))

File: count-the-number-of-consistent-strings1684/leetcode1684.c
#include <stdio.h>

int countConsistentStrings(char * allowed, char ** words, int wordsSize){
    int consistent = wordsSize;
    int diff = 0;
    int wina = 0;
    int w = 0, a=0;
    for (int word=0; word<wordsSize; word++){
        w = 0;
        diff = 0;
        while (words[word][w]){
            a=0;
            wina = 0;
            while (allowed[a]){
                if (words[word][w]==allowed[a]){
                    wina = 1;
                    break;
                }
                a++;
            }
            if (wina==0){
                consistent-=1;
                break;
            }
            w++;
        }
    }
    return consistent;
}

int main(){
    char * allowed = "ab";
    char * words[] = {"ad","bd","aaab","baa","badab"};
    int wordsSize = 5;
    printf("%d", countConsistentStrings(allowed, words, wordsSize));
    return 0;
}